internal/controller/config/config_test.go

```diff
@@ -0,0 +1,126 @@
+package config
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLoadConfigMatchValues(t *testing.T) {
+	getwd, err := os.Getwd()
+	assert.Nil(t, err, "Failed to get working directory")
+	configPath := filepath.Join(getwd, "../../../", "config", "samples")
+	configManager := GetConfigManagerInstance()
+	config := configManager.GetConfig()
+	err = configManager.LoadConfig(func(err error) {
+	}, configPath)
+	assert.Nil(t, err, "Failed to load configuration")
+
+	log.Printf("Loaded Config: %+v", config)
+
+	assert.Equal(t, "example-cluster", config.ClusterName, "ClusterName does not match")
+	assert.Equal(t, uint(60), config.TimeIntervalSec, "TimeIntervalSec does not match")
+
+	assert.NotNil(t, config.RepoCredentials, "RepoCredentials should not be nil")
+
+	assert.NotNil(t, config.RepoCredentials["key1"].HTTPCredential, "HTTPCredential is missing")
+	assert.Equal(t, "exampleUser", config.RepoCredentials["key1"].HTTPCredential.Username, "Username for HTTPCredential does not match")
+	assert.Equal(t, "http-creds", config.RepoCredentials["key1"].HTTPCredential.Password.Name, "Password Name for HTTPCredential does not match")
+	assert.Equal(t, "password", config.RepoCredentials["key1"].HTTPCredential.Password.Key, "Password Key for HTTPCredential does not match")
+
+	assert.NotNil(t, config.RepoCredentials["key2"].SSHCredential, "SSHCredential is missing")
+	assert.Equal(t, "ssh-creds", config.RepoCredentials["key2"].SSHCredential.SSHKey.LocalObjectReference.Name, "SSHKey Name for SSHCredential does not match")
+	assert.Equal(t, "sshKey", config.RepoCredentials["key2"].SSHCredential.SSHKey.Key, "SSHKey Key for SSHCredential does not match")
+
+	assert.NotNil(t, config.RepoCredentials["key3"].TLS, "TLS is missing")
+	assert.True(t, config.RepoCredentials["key3"].TLS.InsecureSkipVerify, "insecureSkipVerify for TLS does not match")
+	assert.Equal(t, "ca-cert-secret", config.RepoCredentials["key3"].TLS.CACertSecret.Name, "CACertSecret Name for TLS does not match")
+	assert.Equal(t, "ca.crt", config.RepoCredentials["key3"].TLS.CACertSecret.Key, "CACertSecret Key for TLS does not match")
+	assert.Equal(t, "cert-secret", config.RepoCredentials["key3"].TLS.CertSecret.Name, "CertSecret Name for TLS does not match")
+	assert.Equal(t, "tls.crt", config.RepoCredentials["key3"].TLS.CertSecret.Key, "CertSecret Key for TLS does not match")
+	assert.Equal(t, "key-secret", config.RepoCredentials["key3"].TLS.KeySecret.Name, "KeySecret Name for TLS does not match")
+	assert.Equal(t, "tls.key", config.RepoCredentials["key3"].TLS.KeySecret.Key, "KeySecret Key for TLS does not match")
+}
+
+// to verify this test run with go test -race ./... it  won't give a race condition as we have used mutex.RwLock
+// in onConfigChange in LoadConfig
```

Comment:
neat idea to test this for the race condition. I wonder if we should in that case update our [Makefile](https://github.com/numaproj-labs/numaplane/blob/main/Makefile#L75) so we have this flag

Reply:
[Here's](https://github.com/open-policy-agent/opa/issues/2388) somebody else who thought about the same thing but decided against it due to it slowing the build down. I wonder if maybe we should add it, and if we have an issue with build time we could remove it. 

(Alternative would be to either add a different Makefile target `make test-with-race` or something, or alternatively have a `TEST_FLAGS` variable that could be set when running, i.e.:
```
        test: manifests generate fmt vet envtest ## Run tests.
	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test -v ${TEST_FLAGS} ./... -coverprofile cover.out
```